shared/tree-sitter-extractor/src/extractor/mod.rs

```diff
@@ -85,41 +95,77 @@ pub fn populate_parent_folders(
     }
 }
 
-fn location(
+#[derive(Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Hash)]
+pub struct Location {
+    pub start_line: usize,
+    pub start_column: usize,
+    pub end_line: usize,
+    pub end_column: usize,
+}
+
+/** Get the label for the given location, defining it a global ID if it doesn't exist yet. */
+fn global_location(
     writer: &mut trap::Writer,
     file_label: trap::Label,
-    start_line: usize,
-    start_column: usize,
-    end_line: usize,
-    end_column: usize,
+    location: Location,
 ) -> trap::Label {
     let (loc_label, fresh) = writer.global_id(&format!(
         "loc,{{{}}},{},{},{},{}",
-        file_label, start_line, start_column, end_line, end_column
+        file_label,
+        location.start_line,
+        location.start_column,
+        location.end_line,
+        location.end_column
     ));
     if fresh {
         writer.add_tuple(
             "locations_default",
             vec![
                 trap::Arg::Label(loc_label),
                 trap::Arg::Label(file_label),
-                trap::Arg::Int(start_line),
-                trap::Arg::Int(start_column),
-                trap::Arg::Int(end_line),
-                trap::Arg::Int(end_column),
+                trap::Arg::Int(location.start_line),
+                trap::Arg::Int(location.start_column),
+                trap::Arg::Int(location.end_line),
+                trap::Arg::Int(location.end_column),
             ],
         );
     }
     loc_label
 }
 
+/** Get the label for the given location, creating it as a fresh ID if we haven't seen the location
+ * yet for this file. */
+fn fresh_location(
```

Comment:
Yeah, the term 'fresh' tends to be a bit overloaded. I just wanted to indicate that, when it does define the label the first time, it uses a fresh ID (`*`) rather than a global ID (`@"loc..."`). But I'm happy to rename it to `location_label`.